unity-renderer/Assets/Rendering/Culling/CullingObjectsTracker.cs

```diff
@@ -1,144 +1,192 @@
+            usingAnimations = null;
+            PoolManager.i.OnGet -= MarkDirty;
+            Rendereable.OnRendereableAdded -= OnRendereableAdded;
+        }
+
+        /// <summary> Sets the dirty flag to true to make PopulateRenderersList retrieve all the scene objects on its next call. </summary>
+        public void MarkDirty() => dirty = true;
+
+        /// <summary> Returns true if dirty. </summary>
+        public bool IsDirty() => dirty;
+
+        /// <summary> Returns the renderers list. </summary>
+        public IReadOnlyList<Renderer> GetRenderers() => usingRenderers;
+
+        /// <summary> Returns the skinned renderers list. </summary>
+        public IReadOnlyList<SkinnedMeshRenderer> GetSkinnedRenderers() => usingSkinnedRenderers;
+
+        /// <summary> Returns the animations list. </summary>
+        public Animation[] GetAnimations() => usingAnimations;
+
+
+        private void ForceCalculateRenderers()
+        {
+            ForceCleanRendereables();
+
+            usingRenderers = new List<Renderer>();
+            foreach (Rendereable rendereable in rendereables)
             {
-                if (renderer == null)
+                if (!rendereable.container.activeInHierarchy)
                     continue;
 
-                if (amount >= CullingControllerSettings.MAX_POPULATING_ELEMENTS_PER_FRAME)
+                foreach (Renderer renderer in rendereable.renderers)
                 {
-                    yield return null;
-                    amount = 0;
-                }
-                amount++;
+                    if (renderer == null || !renderer.gameObject.activeInHierarchy)
+                        continue;
 
-                if ((((1 << renderer.gameObject.layer) & ignoredLayersMask) == 0)
-                    && !(renderer is ParticleSystemRenderer))
-                {
-                    if (renderer is SkinnedMeshRenderer)
-                        skinnedRenderersList.Add((SkinnedMeshRenderer)renderer);
-                    else
-                        renderersList.Add(renderer);
+                    if (((1 << renderer.gameObject.layer) & ignoredLayersMask) == 0)
+                        usingRenderers.Add(renderer);
                 }
             }
-            renderers = renderersList;
-            skinnedRenderers = skinnedRenderersList;
-
-            yield return null;
+        }
 
-            animations = Object.FindObjectsOfType<Animation>();
+        private void ForceCalculateSkinnedRenderers()
+        {
+            usingSkinnedRenderers = new List<SkinnedMeshRenderer>(detectedSkinnedRenderers);
+            foreach (SkinnedMeshRenderer skinnedRenderer in detectedSkinnedRenderers)
+            {
+                if (skinnedRenderer == null)
+                    continue;
 
-            dirty = false;
+                if (((1 << skinnedRenderer.gameObject.layer) & ignoredLayersMask) == 0)
+                    usingSkinnedRenderers.Add(skinnedRenderer);
```

Comment:
Not so sure about merging the two ifs, the pattern of "if check, then ignore, if check, then ignore, ..., in the end: do the work" is a little easier to read than an if with many statements (imho, it's very subjective)

I would use a variable to clarify what the bitwise op is doing. 
```suggestion
                bool matchesIgnoredLayer = (ignoredLayersMask & (1 << skinnedRenderer.gameObject.layer)) != 0
                if (matchesIgnoredLayer)
                    continue;
                usingSkinnedRenderers.Add(skinnedRenderer);
```

Reply:
I am using the first with the method  ShouldBeIgnored(skinnedRenderer) for more clarity